gronx.go

```diff
@@ -70,20 +73,28 @@ func (g *Gronx) IsDue(expr string, ref ...time.Time) (bool, error) {
 }
 
 // Segments splits expr into array array of cron parts.
+// If expression contains 5 parts or 6th part is year like, it prepends a second.
 // It returns array or error.
 func Segments(expr string) ([]string, error) {
 	segs := normalize(expr)
-	if len(segs) < 5 || len(segs) > 6 {
-		return []string{}, errors.New("expr should contain 5-6 segments separated by space")
+	slen := len(segs)
+	if slen < 5 || slen > 7 {
+		return []string{}, errors.New("expr should contain 5-7 segments separated by space")
+	}
+
+	// Prepend second if required
+	prepend := slen == 5 || (slen == 6 && yearRe.MatchString(segs[5]))
```

Comment:
I would say we should not prepend 0 in case there're 6 segments and 6th segment is year it means the expression is invalid.

Reply:
got it, this is for BC reasons. because as is gronx treated 6th segment as year ([ref](https://github.com/adhocore/gronx/blob/7ffa158accaec7b239cea60c01bc8fba0c4ee5c8/README.md#cron-expression))